transport/transport-backend-cct/src/main/java/com/google/android/datatransport/cct/CctTransportBackend.java

```diff
@@ -282,14 +282,18 @@ public BackendResponse send(BackendRequest request) {
     // CCT backend supports 2 different endpoints
     // We route to CCT backend if extras are null and to LegacyFlg otherwise.
     // This (anti-) pattern should not be required for other backends
-    final String apiKey =
-        request.getExtras() == null ? null : LegacyFlgDestination.decodeExtras(request.getExtras());
+    URL actualEndPoint = endPoint;
+    LegacyFlgDestination destination = LegacyFlgDestination.parseFromExtras(request.getExtras());
+    String destinationEndPoint = destination.getEndPoint();
+    if (destinationEndPoint != null && !destinationEndPoint.isEmpty()) {
+      actualEndPoint = parseUrlOrThrow(destination.getEndPoint());
+    }
 
     try {
       HttpResponse response =
           retry(
               5,
-              new HttpRequest(endPoint, requestBody, apiKey),
+              new HttpRequest(actualEndPoint, requestBody, destination.getAPIKey()),
```

Comment:
since this code is shared between lflg and cct backends, doesn't it mean that we will end up always sending the API key to cct even though it's not needed there?

Reply:
You're right.  Fixed.